runhouse/resources/hardware/cluster.py

```diff
@@ -239,11 +240,25 @@ def delete_configs(self):
 
         super().delete_configs()
 
+    def _should_save_creds(self, folder: str = None):
+        import runhouse as rh
+        from runhouse.resources.secrets import Secret
+
+        local_default_folder = folder or rh.configs.username
+        # if not self.rns_address => we are saving the cluster first time in den
+        # else, need to check if the username of the current saver is included in the rns_address.
+        should_save_creds = (
+            not self.rns_address
+            or local_default_folder in self.rns_address
+            or TEST_ORG not in self.rns_address
```

Comment:
this feels like a bit of a patch to add `TEST_ORG`, should avoid having custom logic like this that's really only relevant for internal testing